examples/pennylane_run_rotoselect.py

```diff
@@ -0,0 +1,431 @@
+# converges on the minimum after the first cycle for this simple circuit.
+
+params = init_params.copy()
+opt = qml.GradientDescentOptimizer(stepsize=0.5)
+costs_grad_desc = []
+for i in range(n_steps):
+    costs_grad_desc.append(cost(params))
+    params = opt.step(cost, params)
+
+
+# plot cost function optimization using the 2 techniques
+import matplotlib.pyplot as plt
+
+steps = np.arange(0, n_steps)
+fig, (ax1, ax2) = plt.subplots(1, 2, figsize=(7, 3))
+plt.subplot(1, 2, 1)
+plt.plot(steps, costs_grad_desc, "o-")
+plt.title("grad. desc.")
+plt.xlabel("steps")
+plt.ylabel("cost")
+plt.subplot(1, 2, 2)
+plt.plot(steps, costs_rotosolve, "o-")
+plt.title("rotosolve")
+plt.xlabel("cycles")
+plt.ylabel("cost")
+plt.tight_layout()
+plt.show()
+
+
+##############################################################################
+# Cost function surface for circuit ansatz
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Now, we plot the cost function surface for later comparison with the surface generated
+# by learning the circuit structure. It is apparent that, based on the circuit structure
+# chosen above, the cost function does not depend on the angle parameter :math:`\theta_2`
+# for the rotation gate :math:`R_y`. As we will show in the following sections, this independence is not true
+# for alternative gate choices.
+
+from matplotlib import cm
+from matplotlib.ticker import MaxNLocator
+from mpl_toolkits.mplot3d import Axes3D
+
+fig = plt.figure(figsize=(6, 4))
+ax = fig.gca(projection="3d")
+
+X = np.linspace(-4.0, 4.0, 40)
+Y = np.linspace(-4.0, 4.0, 40)
+xx, yy = np.meshgrid(X, Y)
+Z = np.array([[cost([x, y]) for x in X] for y in Y]).reshape(len(Y), len(X))
+surf = ax.plot_surface(xx, yy, Z, cmap=cm.coolwarm, antialiased=False)
+
+ax.set_xlabel(r"$\theta_1$")
+ax.set_ylabel(r"$\theta_2$")
+ax.zaxis.set_major_locator(MaxNLocator(nbins=5, prune="lower"))
+
+plt.show()
+
+##############################################################################
+# Rotoselect
+# ----------
+#
+# .. figure:: ../../examples/figures/rotoselect_structure.png
+#    :scale: 65%
+#    :align: center
+#    :alt: rotoselect_structure
+#
+# |
+# We now implement the Rotoselect algorithm to learn a good selection of gates to minimize
+# our cost function. The structure is similar to the original ansatz, but the generators of rotation are
+# selected from the set of Pauli gates :math:`P_d \in \{X,Y,Z\}` as shown in the figure above. For example,
+# :math:`U(\theta,Z) = R_z(\theta)`.
+#
+# Creating a quantum circuit with variable gates
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# First, we set up a quantum circuit with a similar structure to the one above, but
+# instead of fixed rotation gates :math:`R_x` and :math:`R_y`, we allow the gates to be specified with the
+# ``generators`` keyword. A helper method ``RGen`` returns the correct unitary gate according to the
+# rotation specified by an element of ``generators``.
```

Comment:
`element of ``generators`` `? make it easier to understand for the user by specifying its structure:

is `generators = ["X", "Y", "Z"]`  ?

Reply:
No generators is the keyword argument that specifies the generators to be used in the circuit. Since there are only 2 rotation gates in this circuit, the possibilities are just ['X','Y'], ['X','Z'], ['Y','X'], and so on. Generators is a keyword argument so it doesn't have a fixed structure: it's an input from the user.

Reply:
yes, but this only becomes clearer later on in the code. maybe add an example or a few words about what could be contained in this argument.

Reply:
Okay got it

Reply:
Added an example